dandiapi/api/services/embargo/utils.py

```diff
@@ -77,8 +78,27 @@ def _delete_zarr_object_tags(client: S3Client, zarr: str):
                 raise AssetTagRemovalError('Some zarr files failed to remove tags', blobs=failed)
 
 
+def _remove_dandiset_manifest_tags(client: S3Client, dandiset: Dandiset):
+    version = dandiset.draft_version
+
+    paths = all_manifest_filepaths(version)
+    logger.info('Removing tags from dandiset %s', dandiset.identifier)
+    for path in paths:
+        try:
+            client.delete_object_tagging(
+                Bucket=settings.DANDI_DANDISETS_BUCKET_NAME,
+                Key=path,
+            )
```

Comment:
I don't think this is correct, long term, because it deletes all tags on the object. That works now since we just use one tag  (`embargoed`) but if we decide to use others for other purposes, this may cause bugs in the future.

I think the correct thing is to use `get_object_tagging()`, remove the `embargoed` tag, then `put_object_tagging()`.

Reply:
While I agree, the reason I didn't worry about that here is because that assumption (that the only tag applied to an S3 object is the embargoed tag) is already made in other parts of the code, as well as the CLI (see https://github.com/dandi/dandi-cli/pull/1685). Changing those other parts would be out of scope for this PR, and I've tested that this PR operates as expected. I can make a follow up PR to change everything in a backwards compatible way.

Reply:
#2517